packages/react/dialog/src/Dialog.tsx

```diff
@@ -169,7 +177,15 @@ const DialogOverlayImpl = React.forwardRef<DialogOverlayImplElement, DialogOverl
     const { __scopeDialog, ...overlayProps } = props;
     const context = useDialogContext(OVERLAY_NAME, __scopeDialog);
     return (
-      <Primitive.div data-state={getState(context.open)} {...overlayProps} ref={forwardedRef} />
+      <RemoveScroll as={Slot} allowPinchZoom={context.allowPinchZoom}>
```

Comment:
While we are making changes here... I noticed recently that `RemoveScroll` already has a `forwardProps` prop that does what `Slot` does. Soooo, we could remove the `Slot` dependency in this package if we used that instead.

Reply:
Ha cool, I'll try that then.